### PR TITLE
Change JsDate.ToString to follow Date prototype logic

### LINKED ISSUE #1424: cfg.LocalTimeZone not work
Jint Version: 3.0.0-beta-2043

this is my code 

```C#
            var siteService = serviceProvider.GetRequiredService<ISiteService>();
            var siteSettings = siteService.GetSiteSettingsAsync().Result;
            var siteTimeZone = TimeZoneInfo.FindSystemTimeZoneById(siteSettings.TimeZoneId);
            var engine = new Engine(options =>
            {
                //TODO: 更新 Jint 版本后可移除
                options.Interop.TrackObjectWrapperIdentity = false;
                options.AddExtensionMethods(typeof(FreeSQLExtensions));
                options.TimeZone= siteTimeZone;
                options.LocalTimeZone(siteTimeZone);//"Asia/Shanghai"
            });
```
input : new Date("2022-1-1")

result : "2021-12-31T16:00:00Z"

Is it because my configuration method is wrong?

Thanks !

### PR DESCRIPTION
**Pull request by lahma**
fixes #1424

### CODE REVIEW

File: Jint/Native/JsDate.cs [hunk 1]
```diff
@@ -56,6 +56,6 @@ public override string ToString()
             return "Infinity";
         }
 
-        return ToDateTime().ToString("ddd MMM dd yyyy HH:mm:ss 'GMT'zzz", CultureInfo.InvariantCulture);
+        return _engine.Realm.Intrinsics.Date.PrototypeObject.ToDateString(DateValue).ToString();
```
Comment (sebastienros) [inline Jint/Native/JsDate.cs:59]: do other JS object follow this pattern?

Comment (sebastienros) [reply Jint/Native/JsDate.cs:59]: meaning JsNumber, ...

Comment (lahma) [reply Jint/Native/JsDate.cs:59]: well the primitives are quite limited with toString, should we just ignore this - don't you like the behavior?

Comment (lahma) [reply Jint/Native/JsDate.cs:59]: Well JsNumber does follow JS semantincs:

```
    public override string ToString()
    {
        return TypeConverter.ToString(_value);
    }
```

Comment (lahma) [reply Jint/Native/JsDate.cs]: Changed to use `TypeConverter` here too

**Review by sebastienros — APPROVED** [review]

**Comment by sebastienros** [comment]
Not sure we need the new dependency. I love the library, use it in Fluid also, but in this case we just need a culture info, we can just use a string that is known from both linux and windows in dotnet core.

**Pull request merged**